### PR TITLE
Add basic CRUD operations for Harbor users

### PR DESCRIPTION
**Pull request by dkorittki**
Add a subclient with create/get/delete/update methods for Harbor-user related operations

### CODE REVIEW

File: scripts/swagger-gen.sh [hunk 1]
```diff
@@ -29,6 +29,11 @@ swagger_operations+=("GetProjectsProjectID")
 swagger_operations+=("PutProjectsProjectID")
 swagger_operations+=("DeleteProjectsProjectID")
 
+swagger_operations+=("GetUsers")
+swagger_operations+=("PostUsers")
+swagger_operations+=("PutUsersUserterminaID")
```
Comment (elenz97) [inline scripts/swagger-gen.sh:34]: Please double check this, as `PutUsersUserterminaID` does not sound correct. Also, I do not see any usage of it.

File: user.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	"github.com/mittwald/goharbor-client/api/v1.10.0/model"
+)
+
+const (
+	ErrUserNotFound     = "user not found on server side"
+	ErrUserBadRequest   = "Unsatisfied with constraints of the user creation/modification."
+	ErrUserMismatch     = "id/name pair not found on server side"
+	ErrUserAlreadyExist = "User with this username already exists."
+	ErrUserInvalidID    = "Invalid user ID."
+)
```
Comment (elenz97) [inline user.go:20]: Personally, I think lower-casing these and removing any punctuation will look better in the end.
```suggestion
const (
	ErrUserNotFound     = "user not found on server side"
	ErrUserBadRequest   = "unsatisfied with constraints of the user creation/modification"
	ErrUserMismatch     = "id/name pair not found on server side"
	ErrUserAlreadyExist = "user with this username already exists"
	ErrUserInvalidID    = "invalid user ID"
)
```
Maybe add comments to each const definition as well.

File: scripts/swagger-gen.sh (hunk 1 above)
Comment (dkorittki) [reply scripts/swagger-gen.sh]: Yes that was a typo. Should have been `PutUsersUserID`. That typo sneaked in after a generated the code, so I only had to fix it in this script file.

File: user.go (hunk 2 above)
Comment (dkorittki) [reply user.go]: Fixed

**Pull request merged**